src/Symfony/Component/OptionsResolver/OptionsResolver.php

```diff
@@ -403,6 +410,37 @@ public function getDefinedOptions(): array
         return array_keys($this->defined);
     }
 
+    /**
+     * Defines nested options.
+     *
+     * @param \Closure(self $resolver, Options $parent): void $nested
+     *
+     * @return $this
+     */
+    public function setNested(string $option, \Closure $nested): static
+    {
+        if ($this->locked) {
+            throw new AccessException('Nested options cannot be defined from a lazy option or normalizer.');
+        }
+
+        $reflection = new \ReflectionFunction($nested);
+        $params = $reflection->getParameters();
+
+        if (!isset($params[0]) || !($type = $params[0]->getType()) instanceof \ReflectionNamedType || self::class !== $type->getName() || (isset($params[1]) && (!($type = $params[1]->getType()) instanceof \ReflectionNamedType || Options::class !== $type->getName()))) {
+            throw new InvalidArgumentException(\sprintf('Argument 2 passed to "%s()" must be a \Closure where the first argument is of type "%s" and the second argument is of type "%s".', __METHOD__, __CLASS__, Options::class));
```

Comment:
I wouldn't do this check. It's going to slow down the call.

Reply:
Agree, removed.